walk/config.go

```diff
@@ -56,6 +61,17 @@ func (wc *walkConfig) isExcluded(rel, base string) bool {
 	return matchAnyGlob(wc.excludes, path.Join(rel, base))
 }
 
+func (wc *walkConfig) isGitIgnored(rel, base string) bool {
+	// .gitignore
+	if wc.gitignoreEnabled && wc.gitignore != nil {
+		if wc.gitignore.isIgnored(strings.Split(path.Join(rel, base), "/")) {
```

Comment:
is it possible to make this work by iterating segments instead of needing to materialize them with a `Split`?

Reply:
You mean maintaining an array as we recurse instead of having to do Split+Join over and over?

Reply:
I think I've got that going...

Reply:
Nice!